File: app/src/main/java/openfoodfacts/github/scrachx/openfood/models/Question.java
package openfoodfacts.github.scrachx.openfood.models;

import com.fasterxml.jackson.annotation.JsonProperty;

import java.io.Serializable;

public class Question implements Serializable {

    private static final long serialVersionUID = 1L;

    @JsonProperty("barcode")
    private String code;
    @JsonProperty("type")
    private String type;
    @JsonProperty("value")
    private String value;
    @JsonProperty("question")
    private String question;
    @JsonProperty("insight_id")
    private String insightId;
    @JsonProperty("insight_type")
    private String insightType;
    @JsonProperty("source_image_url")
    private String sourceImageUrl;
    @JsonProperty("image_url")
    private String imageUrl;


    public String getCode() {
        return code;
    }

    public String getType() {
        return type;
    }

    public String getValue() {
        return value;
    }

    public String getQuestion() {
        return question;
    }

    public String getInsightId() {
        return insightId;
    }

    public String getInsightType() {
        return insightType;
    }

    public String getSourceImageUrl() {
        return sourceImageUrl;
    }

    public String getImageUrl() {
        return imageUrl;
    }
}

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/models/Question.kt
package openfoodfacts.github.scrachx.openfood.models

import com.fasterxml.jackson.annotation.JsonIgnore
import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import com.fasterxml.jackson.annotation.JsonProperty
import java.io.Serializable

@JsonIgnoreProperties(ignoreUnknown = true)
data class Question(
    @JsonProperty("barcode") val code: String? = null,
    @JsonProperty("type") val type: String? = null,
    @JsonProperty("value") val value: String? = null,
    @JsonProperty("question") val questionText: String? = null,
    @JsonProperty("insight_id") val insightId: String? = null,
    @JsonProperty("insight_type") val insightType: String? = null,
    @JsonProperty("source_image_url") val sourceImageUrl: String? = null,
    @JsonProperty("image_url") val imageUrl: String? = null,
) : Serializable {
    @JsonIgnore
    fun isEmpty() = questionText.isNullOrEmpty()

    companion object {
        private const val serialVersionUID = 1L
    }
}